controllers/storagecluster/backingstorageclasses.go

```diff
@@ -0,0 +1,162 @@
+package storagecluster
+
+import (
+	"fmt"
+	"reflect"
+
+	configv1 "github.com/openshift/api/config/v1"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v1"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	v1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	backingStorageClassLabel = "ocs.openshift.io.backingstorageclass"
+)
+
+type backingStorageClasses struct{}
+
+// ensureCreated ensures that backing storageclasses are created for the StorageCluster
+func (obj *backingStorageClasses) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) (reconcile.Result, error) {
+
+	platform, err := r.platform.getPlatform(r.Client)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	existingBackingStorageClasses, err := getBackingStorageClasses(r)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	for i := range sc.Spec.BackingStorageClasses {
+		bsc := &sc.Spec.BackingStorageClasses[i]
+		err := createOrUpdateBackingStorageclass(r, bsc, platform)
+		if err != nil {
+			r.Log.Error(err, "Failed to create StorageClass.", "StorageClass", bsc.Name)
+			continue
+		}
+		findAndDeleteStorageClassFromList(existingBackingStorageClasses, bsc.Name)
+	}
+
+	for i := range existingBackingStorageClasses.Items {
+		storageClass := &storagev1.StorageClass{}
+		storageClass.Name = existingBackingStorageClasses.Items[i].Name
+		if err := r.Client.Delete(r.ctx, storageClass); err != nil && !errors.IsNotFound(err) {
+			r.Log.Error(err, "Unable to delete BackingStorageClass.", "Name", storageClass.Name)
+		}
+	}
```

Comment:
1. see comment above
2. use `r.delete` instead of r.Client.delete
3. Don't stop deleting because of a single error.

```suggestion
	for _, storageClass := range existingBackingStorageClasses.Items {
		if storageClass == nil {
		
		} 
		if err := r.Client.Delete(r.ctx, storageClass); err != nil && !errors.IsNotFound(err) {
			r.Log.Error(err, "Unable to delete BackingStorageClass.", "Name", storageClass.Name)
		}
	}
```